packages/styled/types/tests.tsx

```diff
@@ -217,3 +217,43 @@ const Input5 = styled.input`
   // $ExpectError
   ;<StyledCompWithoutAs as={Section} />
 }
+
+{
+  type StyledOptsParameterized = StyledOptions<{ a: string; b: number }>
+
+  const styledOpts0: StyledOptions = {
+    label: 'foo',
+    target: 'bar',
+    shouldForwardProp: p => true
+  }
+  const styledOpts1: StyledOptsParameterized = {
+    label: 'foo',
+    target: 'bar',
+    shouldForwardProp: (p: 'a' | string) => true
```

Comment:
what was the intention here? `'a' | string` is an equivalent of just `string`

Reply:
I tried to verify that `p` value type is properly extracted as a string key union. I'll modify it to make it a bit clearer.